HHS/simpler-grants-gov
# [Issue 2616] show correct date opportunity data with collapsable text

## Summary
Fixes #2616

### Time to review: __20 mins__

## Changes proposed

Functionally:
* replaces the hardcoded close date warning message in the opportunity status widget with the the close date description from the opportunity summary data
* handles long summary descriptions and long close date descriptions by collapsing the description after a certain number of characters
    * for close date description: if description is over 150 characters any characters after 120 are collapsed
    * for close date description: if description is over 750 characters any characters after 600 are collapsed

Behind the scenes:
* updates the ContentDisplayToggle component to be useful in this use case with flexibility around horizontal alignment and the ability to display hidden content above the button when expanded
* creates utility function to handle splitting a string, such as the summary description, that may contain markup that we want to not break by splitting at the wrong spot
* updates local seed data with two records containing long summary and close date descriptions to ease local testing of the cases handled in this PR


## Context for reviewers

Note that at this point we should be running e2e tests. Follow up ticket created here https://github.com/HHS/simpler-grants-gov/issues/2715

### Test steps

1. re-seed your local environment to bring in the new examples with long descriptions `make db-seed-local`
2. visit the opportunity page for one of the newly seeded examples. This will probably be 32, but check on your local db to see
3. _VERIFY_: close date description text in the status widget comes from the close date description of the opportunity summary, not hardcoded string
4. _VERIFY_: close date description is cut off at the correct point, with remaining content hidden by toggle
5. _VERIFY_: summary description is cut off at the correct point, with remaining content hidden by toggle
6. _VERIFY_: both toggles work to display hidden content

## Additional information

<img width="1299" alt="Screenshot 2024-11-04 at 11 35 18 AM" src="https://github.com/user-attachments/assets/65515f15-f19a-4b19-b7fd-a376b85ff4e5">
